spyderlib/plugins/shortcuts.py

```diff
@@ -789,8 +789,9 @@ def setup_page(self):
         self.setTabOrder(self.finder, self.reset_btn)
 
         # Signals and slots
+        #    Qt5 'dataChanged' has 3 parameters
         self.table.proxy_model.dataChanged.connect(
-                     lambda i1, i2, opt='': self.has_been_modified(opt))
+                     lambda i1, i2, roles=[], opt='': self.has_been_modified(opt))
         self.reset_btn.clicked.connect(self.reset_to_default)

```

Comment:
Hi @Qlogin, thanks for working on this.

I think that to make the code much clearer to future contributors we could check the current API in use and then feed the correct connection

Like

```python
from spyderlib.qt import PYQT5

...

if PYQT5:
    #    Qt5 'dataChanged' has 3 parameters
    self.table.proxy_model.dataChanged.connect(lambda i1, i2, roles=[], opt='': self.has_been_modified(opt))
else:
    self.table.proxy_model.dataChanged.connect(lambda i1, i2, opt='': self.has_been_modified(opt))